### PR TITLE
Validate total number of ballots and total number of signatures

### LINKED ISSUE #290: Calculations Based Triggers: Validate the total number of ballots at the station
### User story
---
The total number of received ballots at the station should equal the total of the ballots inside the box plus ballots outside the box at the end of the polling.

### Implementation:
---

- Calculate number of ballots inside the box by summing, valid and invalid votes, and unstamped ballots. 
- Calculate number of ballots outside the box by summing, unused, discarded/cancelled and spoiled ballots.
- Check if the total number of ballots received is equal to the sum of number of ballots inside the box and number of ballots outside the box.

### LINKED ISSUE #291: Calculations Based Triggers: Validate the total number of signatures with ballots inside the box
### User story
----
The total number of signatures with ballots inside the box: the total number of signatures on the voter list for the voters who voted should equal the number of the ballots found in the ballot box after polling plus Discarded ballots. 

### Implementation 
---

- Get total number of signatures for recon form.
- Calculate ballots inside the box (valid + invalid + unstamped) + Discarded/Cancelled.
- Check that total number of signatures is equal to the number of ballots inside the box.

### PR DESCRIPTION
**Pull request by JohnMwashuma**
### Changes / Features implemented

- Add quarantine check for validating total number of ballots
- Add quarantine check for validating total number of signatures

### Pending Work

- [x] Add/Update tests

### Steps taken to verify this change does what is intended

- [x] Add/Updated tests

### Side effects of implementing this change

Closes #290 
Closes #291

### CODE REVIEW

File: tally_ho/apps/tally/models/reconciliation_form.py [hunk 1]
```diff
@@ -77,5 +77,29 @@ def number_ballots_expected(self):
                 self.number_unstamped_ballots -
                 self.number_invalid_votes)
 
+    @property
+    def number_ballots_inside(self):
```
Comment (pld) [inline tally_ho/apps/tally/models/reconciliation_form.py:81]: the word "inside" is ambiguous to me, what does that mean? Is it received in any ballot box?

File: tally_ho/libs/verify/quarantine_checks.py [hunk 2]
```diff
@@ -97,6 +101,59 @@ def pass_tampering(result_form):
     return diff <= scaled_tolerance
 
 
+def pass_ballots_number_validation(result_form):
+    """Validate that the total number of received ballots equals the
+    total of the ballots inside plus ballots outside.
```
Comment (pld) [inline tally_ho/libs/verify/quarantine_checks.py:106]: what are ballots `outside`?

File: tally_ho/apps/tally/models/reconciliation_form.py (hunk 1 above)
Comment (JohnMwashuma) [reply tally_ho/apps/tally/models/reconciliation_form.py:81]: I will update this, should read `number_ballots_inside_the_box` instead

File: tally_ho/libs/verify/quarantine_checks.py (hunk 2 above)
Comment (JohnMwashuma) [reply tally_ho/libs/verify/quarantine_checks.py:106]: Same here, `ballots inside the box` and `ballots outside the box`.

Comment (JohnMwashuma) [reply tally_ho/libs/verify/quarantine_checks.py]: Done

File: tally_ho/apps/tally/models/reconciliation_form.py (hunk 1 above)
Comment (JohnMwashuma) [reply tally_ho/apps/tally/models/reconciliation_form.py]: Done

**Pull request merged**